lib/gridstore/src/bitmask/mod.rs

```diff
@@ -379,27 +377,27 @@ impl Bitmask {
     ) {
         let page_start = self.range_of_page(page_id).start;
 
-        let mut lowest_offset = usize::MAX;
-        let mut highest_offset = page_start;
+        let est_num_ranges = local_block_ranges.size_hint().1.unwrap_or(1);
+        let mut dirty_regions = AHashSet::with_capacity(est_num_ranges);
+
         for range in local_block_ranges {
             let bitmask_range = (range.start + page_start)..(range.end + page_start);
             self.bitslice[bitmask_range.clone()].fill(used);
 
-            lowest_offset = lowest_offset.min(bitmask_range.start);
-            highest_offset = highest_offset.max(bitmask_range.end);
-        }
+            let start_region_id =
+                (bitmask_range.start / self.config.region_size_blocks) as RegionId;
+            let end_region_id =
+                ((bitmask_range.end - 1) / self.config.region_size_blocks) as RegionId;
```

Comment:
Are we sure about the `-1` here?

Just curious because `bitmask_range = (range.start + page_start)..(range.end + page_start)` is exclusive already.

Reply:
This is a bit confusing, yes. I did this so that we could use an inclusive range below (`start_region_id..=end_region_id`). But a more readable form might be using `div_ceil` with an exclusive range

Reply:
Thanks for having a look at it, and for the change!